### ISSUE TITLE
Better documentation

### ISSUE THREAD

**Issue by credmond** [TestFX/TestFX#793]
## Expected Behavior
Better documentation (not JavaDoc), covering various use cases and utility classes. 

## Actual Behaviour
There's very little relevant documentation, but there seems to be a tonne of functionality within TestFX (observed via browsing the code). There's practically no information on the majority of TestFX.

**Comment by svenruppert** [comment]
You are right; better documentation is one target for the TestFX5 version. Step by step, we can add documentation for TestFX4 as well.
